v03_pipeline/lib/reference_datasets/reference_dataset.py

```diff
@@ -15,71 +15,54 @@
 )
 
 DATASET_TYPES = 'dataset_types'
-FILTER = 'filter'
-VERSION = 'version'
-RAW_DATASET_PATH = 'raw_dataset_path'
 ENUMS = 'enums'
-IS_INTERVAL = 'is_interval'
 EXCLUDE_FROM_ANNOTATIONS = 'exclude_from_annotations'
-CUSTOM_SELECT = 'custom_select'
+FILTER = 'filter'
+IS_INTERVAL = 'is_interval'
+SELECT = 'select'
+VERSION = 'version'
+PATH = 'path'
 
 
 class BaseReferenceDataset:
     @classmethod
-    def _for_reference_genome_dataset_type(
+    def for_reference_genome_dataset_type(
         cls,
         reference_genome: ReferenceGenome,
         dataset_type: DatasetType,
-    ) -> list[Union['ReferenceDataset', 'ReferenceDatasetQuery']]:
+    ) -> set[Union['ReferenceDataset', 'ReferenceDatasetQuery']]:
```

Comment:
I changed these to be sets instead of lists for easier downstream set union and symmetric difference